pkg/capacityscheduling/capacity_scheduling_test.go

```diff
@@ -168,6 +168,137 @@ func TestPreFilter(t *testing.T) {
 	}
 }
 
+func TestReserve(t *testing.T) {
+	tests := []struct {
+		name          string
+		pods          []*v1.Pod
+		elasticQuotas map[string]*ElasticQuotaInfo
+		expected      []framework.Code
+	}{
+		{
+			name: "Reserve pods",
+			pods: []*v1.Pod{
+				makePod("t1-p1", "ns1", 50, 0, 0, midPriority, "t1-p1", "node-a"),
+				makePod("t1-p2", "ns2", 50, 0, 0, midPriority, "t1-p2", "node-a"),
+			},
+			elasticQuotas: map[string]*ElasticQuotaInfo{
+				"ns1": {
+					Namespace: "ns1",
+					pods:      sets.String{},
+					Min: &framework.Resource{
+						Memory: 1000,
+					},
+					Max: &framework.Resource{
+						Memory: 2000,
+					},
+					Used: &framework.Resource{
+						Memory: 300,
+					},
+				},
+			},
+			expected: []framework.Code{
+				framework.Success,
+				framework.Success,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var registerPlugins []st.RegisterPluginFunc
+			registeredPlugins := append(
+				registerPlugins,
+				st.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
+				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
+			)
+
+			fwk, err := st.NewFramework(
+				registeredPlugins, "", wait.NeverStop,
+				frameworkruntime.WithPodNominator(testutil.NewPodNominator(nil)),
+				frameworkruntime.WithSnapshotSharedLister(testutil.NewFakeSharedLister(make([]*v1.Pod, 0), make([]*v1.Node, 0))),
+			)
+
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			cs := &CapacityScheduling{
+				elasticQuotaInfos: tt.elasticQuotas,
+				fh:                fwk,
+			}
+
+			state := framework.NewCycleState()
+			for i, pod := range tt.pods {
+				got := cs.Reserve(nil, state, pod, "node-a")
+				if got.Code() != tt.expected[i] {
```

Comment:
Could you please further check whether the pod is reserved correctly in the internal cache (`elasticQuotaInfos`)? (applies to Unreserve as well)

Reply:
Thanks for your comment. Fixed it